### PR TITLE
Network reading failure in IE for operations ExperimentalDetectronPriorGridGenerator and ExperimentalDetectronROIFeatureExtractor

### PR DESCRIPTION
**Pull request by vgavrilo**
Network reading failure in IE for operations `ExperimentalDetectronPriorGridGenerator` and `ExperimentalDetectronROIFeatureExtractor `

Tickets: #47385, #47543

### CODE REVIEW

**Comment by ilyachur** [comment]
Please add new tests which cover this functionality.

**Comment by vgavrilo** [comment]
> Please add new tests which cover this functionality.

Done. Please, see them.

**Review by lazarevevgeny — APPROVED** [review]

**Comment by vgavrilo** [comment]
@mvafin, @iimironov, @evolosen, please, see this PR.

File: ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp [hunk 1]
```diff
@@ -169,3 +169,153 @@ TEST(type_prop, detectron_roi_feature_extractor_intervals)
...
+    struct Shapes
+    {
+        PartialShape input_shape;
+        Dimension channels[4];
+        Dimension first_dims[4];
+    };
+
+    const auto dyn_dim = Dimension::dynamic();
+
+    std::vector<Shapes> shapes = {
```
Comment (ilyachur) [inline ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp:190]: I think will be better to use the standard approach and create parametrized test.

Comment (vgavrilo) [reply ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp:190]: Please, give me link to an example of such `type_prop.*` tests.

Comment (ilyachur) [reply ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp:190]: You can try to use `TEST_P` as a keyword to search in the OpenVINO sources, I think you will get a lot of results or you can use [google](https://www.google.com/search?q=gtest+parameterized+test+example) to find more examples.

Comment (vgavrilo) [reply ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp]: Done.

**Comment by vgavrilo** [comment]
@ilyachur, @iimironov, @evolosen, @mvafin, please, see this PR (https://github.com/openvinotoolkit/openvino/pull/4014).

File: ngraph/test/type_prop/experimental_detectron_prior_grid_generator.cpp [hunk 2]
```diff
@@ -176,3 +176,99 @@ TEST(type_prop, detectron_grid_generator_dynamic_shapes_intervals)
         ASSERT_TRUE(grid_gen->get_output_partial_shape(0).same_scheme(s.ref_out_shape));
     }
 }
+
+TEST(type_prop, detectron_grid_generator_dynamic_shapes_intervals_1)
```
Comment (ilyachur) [inline ngraph/test/type_prop/experimental_detectron_prior_grid_generator.cpp:180]: Please use parametrized test.

File: ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp [hunk 3]
```diff
@@ -169,3 +169,192 @@ TEST(type_prop, detectron_roi_feature_extractor_intervals)
...
+    ASSERT_TRUE(roi->get_output_partial_shape(0).same_scheme(ref_out_shape));
+}
+
+struct ROIFeatureIntervalsTest : ::testing::TestWithParam<ROIFeatureIntervalsTestParams>
+{
+};
+
+TEST_P(ROIFeatureIntervalsTest, detectron_roi_feature_extractor_intervals_1)
+{
+    run_roi_feature_test(GetParam());
```
Comment (ilyachur) [inline ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp:221]: I don't understand why do you need to have an additional method `run_roi_feature_test`? Looks like it is a redundant function.
You can have all code in the test body.

**Review by evolosen — APPROVED** [review]

File: ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp (hunk 3 above)
Comment (vgavrilo) [reply ngraph/test/type_prop/experimental_detectron_roi_feature_extractor.cpp]: Done.

File: ngraph/test/type_prop/experimental_detectron_prior_grid_generator.cpp (hunk 2 above)
Comment (vgavrilo) [reply ngraph/test/type_prop/experimental_detectron_prior_grid_generator.cpp:217]: Fixed.

**Comment by vgavrilo** [comment]
@ilyachur, @lazarevevgeny, @mvafin, @iimironov, please, see this PR (https://github.com/openvinotoolkit/openvino/pull/4014)

File: ngraph/test/type_prop/experimental_detectron_prior_grid_generator.cpp (hunk 2 above)
Comment (ilyachur) [reply ngraph/test/type_prop/experimental_detectron_prior_grid_generator.cpp:217]: As I see you still don't have parametrized test.

File: ngraph/core/src/op/experimental_detectron_prior_grid_generator.cpp [hunk 4]
```diff
@@ -95,8 +95,9 @@ void op::v6::ExperimentalDetectronPriorGridGenerator::validate()
 
     const auto num_batches_featmap = featmap_shape[0];
     const auto num_batches_im_data = im_data_shape[0];
+    const auto batches_intersection = num_batches_featmap & num_batches_im_data;
```
Comment (ilyachur) [inline ngraph/core/src/op/experimental_detectron_prior_grid_generator.cpp:98]: Why do you use logical and operation?

Comment (vgavrilo) [reply ngraph/core/src/op/experimental_detectron_prior_grid_generator.cpp:98]: It is not logical and operation, it is an intersection of intervals that represent dimensions `featmap_shape[0]` and `im_data_shape[0]`.

Comment (ilyachur) [reply ngraph/core/src/op/experimental_detectron_prior_grid_generator.cpp:98]: Sure It a typo, of course I asked about binary and operation.

**Review by ilyachur — APPROVED** [review]

File: ngraph/test/type_prop/experimental_detectron_prior_grid_generator.cpp (hunk 2 above)
Comment (vgavrilo) [reply ngraph/test/type_prop/experimental_detectron_prior_grid_generator.cpp]: Done.

**Review by jane-intel — APPROVED** [review]

**Pull request merged**